### ISSUE TITLE
Follow the Reader fails to install

### ISSUE THREAD

**Issue by distantvale** [dosbox-staging/dosbox-staging#3648]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

Windows 11

### Relevant hardware info

12th Gen Intel, Intel Iris GPU

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

After selecting all options during the installation process, the following error shows up:

![image](https://github.com/dosbox-staging/dosbox-staging/assets/125926308/540c3ef7-61d0-416e-9254-3faa19702d40)

Note: I tried it under DOSBox-X and it installs just fine.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. download the linked game and extract it to a folder
2. create a file named "READER 1" in the same folder (this is critical!)
3. run the install program
4. continue with all default settings

Note that if you don't create the file on step #2, it installs correctly.
Also note that on DOSBox-X this installs just fine with the file created.
The issue is probably related to file name collision but there shouldn't be any - the issue occurs even when installing to a different folder name that the default "READER".

### Download URL of affected game or software

https://www.myabandonware.com/download/uo6-follow-the-reader

### Your configuration

```ini
cputype   = 386_prefetch (tried with pentium_slow as well
xms                   = true
ems                   = true
umb                   = true
(tried with the above also set to false)
machine                     = svga_s3 (tried others as well)
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by johnnovak** [comment]
@distantvale Does thus happen in the previous stable, so 0.80.1 as well? In other words, is this a regression?

**Comment by distantvale** [comment]
@johnnovak tried it now in a clean 0.80, same thing, so doesn't look like a regression.
also nothing in the log.

**Comment by johnnovak** [comment]
> @johnnovak tried it now in a clean 0.80, same thing, so doesn't look like a regression. also nothing in the log.

Thanks for confirming it's a long-standing bug.

**Comment by weirddan455** [comment]
I looked into this a bit.  This error only happens when you create an empty file (0 bytes on the host) that matches the installer's search parameter.  If the file is not empty, there is no error.  Specifically, it looks for these things.

```
2024-05-11 06:58:58.676 | Finding C:\BUG\INSTALL.*
2024-05-11 06:58:58.702 | Finding C:\BUG\READ*.*
2024-05-11 06:58:58.711 | Finding C:\BUG\MR*.EXE
2024-05-11 06:58:58.864 | Finding C:\BUG\SETUP.*
2024-05-11 06:58:58.868 | Finding C:\BUG\DATA\DISK1.*
2024-05-11 06:59:00.568 | Finding C:\BUG\DATA\DISK2.*
2024-05-11 06:59:15.193 | Finding C:\BUG\DATA\DISK3.*
```
I can't find anything that we're doing wrong with empty files.  It actually creates the empty file in the destination directory and *then* throws an error.  It could just be the installer's logic not expecting 0 length files.

Needs testing on real MS-DOS.  It's probably either us or Dosbox-X getting something wrong but which one?  :man_shrugging:

**Comment by johnnovak** [comment]
> Needs testing on real MS-DOS. It's probably either us or Dosbox-X getting something wrong but which one? 🤷‍♂️

Yeah, could be easily tested in PCem/86box.

**Comment by weirddan455** [comment]
Ok, I was able to test in 86box after much fiddling with their settings to get a working MS-DOS 6.22 install.  As best I can tell, Staging matches behavior of real DOS in this case as I was able to get the same error on 86box.

DOSBox-X likely has a bug that makes the installer "work" when it should really be throwing an error because you put a garbage file in its path.

![reader](https://github.com/dosbox-staging/dosbox-staging/assets/8128047/dde009fb-deb0-4975-8e30-50ce762a201f)
